tls/s2n_record_read.c

```diff
@@ -98,16 +98,144 @@ int s2n_record_header_parse(struct s2n_connection *conn, uint8_t * content_type,
     return 0;
 }
 
+int s2n_record_read_decryption_non_aead(struct s2n_connection *conn, const struct s2n_cipher_suite *cipher_suite, struct s2n_session_key *session_key,
```

Comment:
I had an offline discussion with @alexw91 and @raycoll, we decided to pull out s2n_record_read.c and s2n_record_write.c changes for this request. I will submit another pull request just for changes in those 2 files, will address your comment there. :)